packages/commons/src/auth/authorizationMiddleware.ts

```diff
@@ -92,25 +89,16 @@ export const authorizationMiddleware =
       return next();
     } catch (err) {
       const headers = readHeaders(req as Request);
-
       const problem = match<unknown, Problem>(err)
-        .with(P.instanceOf(ProcessError), (error) =>
+        .with(P.instanceOf(ApiError), (error) =>
           makeApiProblem(
-            error.type.code,
-            error.type.httpStatus,
-            error.type.title,
-            error.message,
-            headers?.correlationId
+            new ApiError({
+              ...error,
+              correlationId: headers?.correlationId,
+            })
           )
         )
-        .otherwise(() =>
-          makeApiProblem(
-            `${ErrorTypes.GenericError.code}`,
-            ErrorTypes.GenericError.httpStatus,
-            ErrorTypes.GenericError.title,
-            "Generic error"
-          )
-        );
+        .otherwise(() => makeApiProblem(genericError("Generic error")));
```

Comment:
do you think it's possible to have more details here?

Reply:
I can use a more specific error message